library/L1_Peripheral/lpc17xx/test/system_controller_test.cpp

```diff
@@ -0,0 +1,190 @@
+
+TEST_CASE("Testing LPC176x/5x System Controller", "[lpc17xx-SystemCoontroller]")
+{
+  using PeripheralID = SystemController::PeripheralID;
+
+  // Simulate local version of LPC_SC
+  LPC_SC_TypeDef local_sc;
+  memset(&local_sc, 0, sizeof(local_sc));
+  SystemController::system_controller = &local_sc;
+
+  constexpr uint8_t kPclkSel1StartIndex = 13;
+  constexpr PeripheralID kPeripherals[] = {
+    SystemController::Peripherals::kTimer0,
+    SystemController::Peripherals::kTimer1,
+    SystemController::Peripherals::kUart0,
+    SystemController::Peripherals::kUart1,
+    SystemController::Peripherals::kPwm1,
+    SystemController::Peripherals::kI2c0,
+    SystemController::Peripherals::kSpi,
+    SystemController::Peripherals::kRtc,
+    SystemController::Peripherals::kSsp1,
+    SystemController::Peripherals::kAdc,
+    SystemController::Peripherals::kCan1,
+    SystemController::Peripherals::kCan2,
+    SystemController::Peripherals::kGpio,
+    SystemController::Peripherals::kRit,
+    SystemController::Peripherals::kMotorControlPwm,
+    SystemController::Peripherals::kQuadratureEncoder,
+    SystemController::Peripherals::kI2c1,
+    SystemController::Peripherals::kSsp0,
+    SystemController::Peripherals::kTimer2,
+    SystemController::Peripherals::kTimer3,
+    SystemController::Peripherals::kUart2,
+    SystemController::Peripherals::kUart3,
+    SystemController::Peripherals::kI2c2,
+    SystemController::Peripherals::kI2s,
+    SystemController::Peripherals::kGpdma,
+    SystemController::Peripherals::kEthernet,
+    SystemController::Peripherals::kUsb,
+  };
+
+  SystemController system_controller = SystemController();
+
+  SECTION("Set CPU Clock Frequency")
+  {
+    constexpr uint32_t kInputFrequencyInKhz    = 4'000;
+    constexpr uint32_t kExpectedFrequencyInMhz = 48;
+
+    system_controller.SetSystemClockFrequency(kExpectedFrequencyInMhz);
+
+    CHECK(bit::Read(local_sc.PLL0CON, SystemController::kPllEnableBit) == 0b1);
+    CHECK(bit::Read(local_sc.PLL0CON, SystemController::kPllConnectBit) == 0b1);
+
+    const uint32_t kMultiplier =
+        bit::Extract(local_sc.PLL0CFG, SystemController::MainPll::kMultiplier);
+    const uint32_t kPreDivider =
+        bit::Extract(local_sc.PLL0CFG, SystemController::MainPll::kPreDivider);
+    const uint32_t kCpuDivider =
+        bit::Extract(local_sc.CCLKCFG, SystemController::CpuClock::kDivider);
+    const uint32_t kClockFrequencyInKhz =
+        ((2 * (kMultiplier + 1) * kInputFrequencyInKhz) / (kPreDivider + 1)) /
+        (kCpuDivider + 1);
+
+    CHECK(bit::Extract(local_sc.CLKSRCSEL,
+                       SystemController::Oscillator::kSelect) == 0b00);
+    CHECK(kClockFrequencyInKhz == (48 * 1'000));
+    CHECK(system_controller.GetSystemFrequency() == (48 * 1'000'000));
+  }
+
+  SECTION("Set USB PLL Input Frequency")
+  {
+    using UsbPllInputFrequency = SystemController::UsbPllInputFrequency;
+
+    const UsbPllInputFrequency kInputFrequencies[] = {
```

Comment:
You only use ` using UsbPllInputFrequency = SystemController::UsbPllInputFrequency;` once here. Mine as well just use:
```suggestion
    const  SystemController::UsbPllInputFrequency kInputFrequencies[] = {
```